src/groupeddataframe/grouping.jl

```diff
@@ -145,27 +162,65 @@ determines the shape of the resulting data frame:
 In all cases, the resulting `GroupedDataFrame` contains all the grouping columns in addition
 to those listed above. Note that `f` must always return the same type of object for
 all groups, and (if a named tuple or data frame) with the same fields or columns.
-Returning a single value or a named tuple is significantly faster than
+
+Note that specifying columns via `select` is dramatically faster than `select=nothing`,
+and that returning a single value or a named tuple is significantly faster than
 returning a vector or a data frame.
 
 ### Examples
 
 ```julia
-df = DataFrame(a = repeat([1, 2, 3, 4], outer=[2]),
-               b = repeat([2, 1], outer=[4]),
-               c = randn(8))
-gd = groupby(df, :a)
-map(d -> sum(skipmissing(d[:c])), gd)
+julia> df = DataFrame(a = repeat([1, 2, 3, 4], outer=[2]),
+                      b = repeat([2, 1], outer=[4]),
+                      c = randn(8));
+
+julia> gd = groupby(df, :a);
+
+julia> map(c -> sum(skipmissing(c)), gd; select=:c)
+GroupedDataFrame{DataFrame} with 4 groups based on key: :a
+First Group: 1 row
+│ Row │ a     │ x1       │
+│     │ Int64 │ Float64  │
+├─────┼───────┼──────────┤
+│ 1   │ 1     │ 0.569396 │
+⋮
+Last Group: 1 row
+│ Row │ a     │ x1      │
+│     │ Int64 │ Float64 │
+├─────┼───────┼─────────┤
+│ 1   │ 4     │ 1.40818 │
+
+julia> map(d -> sum(skipmissing(d[:c])), gd) # Slower variant
+GroupedDataFrame{DataFrame} with 4 groups based on key: :a
+First Group: 1 row
+│ Row │ a     │ x1       │
+│     │ Int64 │ Float64  │
+├─────┼───────┼──────────┤
+│ 1   │ 1     │ 0.569396 │
+⋮
+Last Group: 1 row
+│ Row │ a     │ x1      │
+│     │ Int64 │ Float64 │
+├─────┼───────┼─────────┤
+│ 1   │ 4     │ 1.40818 │
 ```
 
 ### See also
 
 `combine(f, gd)` returns a `DataFrame` rather than a `GroupedDataFrame`
 
 """
-function Base.map(f::Union{Function, Type}, gd::GroupedDataFrame)
+function Base.map(f::Union{Function, Type}, gd::GroupedDataFrame;
+                  select::Union{Nothing, ColumnIndices}=nothing)
     if length(gd) > 0
-        idx, valscat = _combine(wrap(f(gd[1])), f, gd)
+        if select === nothing
+            incols = nothing
+        elseif select isa Symbol || select isa Integer
+            incols = (gd.parent[select],)
+        else
+            incols = Tuple(columns(gd.parent[collect(select)]))
```

Comment:
This is not optimal in general but I guess it is good enough as I would assume that this is not the costly operation relative to the whole work that has to be done and `select` is has few elements.